corehq/apps/translations/validator.py

```diff
@@ -35,28 +42,48 @@ def __init__(self, app, uploaded_workbook, lang_to_compare, lang_prefix='default
         self.headers = None
         self.expected_rows = None
         self.lang_prefix = lang_prefix
-        self.lang_cols_to_compare = [self.lang_prefix + self.app.default_language]
         self.default_language_column = self.lang_prefix + self.app.default_language
-        if lang_to_compare in app.langs and lang_to_compare != self.app.default_language:
-            self.lang_cols_to_compare.append(self.lang_prefix + lang_to_compare)
-            target_lang = lang_to_compare
+        self.lang_to_compare = lang_to_compare
+        self.single_sheet = False
+        self._setup()
+
+    def _setup(self):
+        if self.lang_to_compare:
+            # assume its a single sheet workbook if there is a language
+            self.single_sheet = True
+            self._ensure_single_sheet()
+            target_lang = self.lang_to_compare
+            self.lang_cols_to_compare = [self.lang_prefix + self.lang_to_compare]
         else:
             target_lang = self.app.default_language
+            self.lang_cols_to_compare = [self.lang_prefix + self.app.default_language]
         self.app_translation_generator = AppTranslationsGenerator(
             self.app.domain, self.app.get_id, None, self.app.default_language, target_lang,
             self.lang_prefix)
 
+    def _ensure_single_sheet(self):
+        sheet = self.uploaded_workbook.worksheets[0]
+        if not is_single_sheet(sheet.title):
+            raise Exception("Expected single sheet with title %s" % SINGLE_SHEET_NAME)
```

Comment:
How about making this a `BulkAppTranslationsException`?

Reply:
Since we were not catching it, i didn't see the use of creating a custom exception. Should i still?

Reply:
I see your point, it doesn't really matter. Though `BulkAppTranslationsException` already exists, it wouldn't be new.

Reply:
ya i did see the list of pre-existed exceptions when i was writing this, but none of them look apt for "an issue with uploaded sheet" because it has not reached to doing translations yet.